allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -25,16 +25,16 @@
 import torch.distributed as dist  # type: ignore
 import torch.distributions  # type: ignore
 import torch.multiprocessing as mp  # type: ignore
-import torch.optim
-from torch import nn
-from torch import optim
+import torch.optim as optim
```

Comment:
do you know why this is correct versus `from torch import optim` (the latter reads better to me)?

Reply:
I'm not sure actually.

I think, in general, people tend to discourage doing `from module import func` because:
* it pollutes the local namespace and makes it more difficult to figure out where things are defined, and
* it makes it difficult to dynamically reload/mock modules (i.e. if I want to do some debugging where I replace `numpy` with my own custom module, then if I'm doing `np.func` everywhere I can just do `np = my_new_module` whereas if I've actually imported `func` then I would need to manually know to do `func = my_new_module.func`.) 
That all said, I think the above (especially the first point) is somewhat less relevant nowadays when IDEs like pycharm make it so easy to jump to definitions.

This isn't actually what was being flagged by lgtm.com though, it was complaining that we were mixing styles, i.e. doing both
```
import module
from module import x
```
I think this is purely a stylistic thing.